packages/admin-ui/ui/src/domain/product-categories/modals/add-product-category.tsx

```diff
@@ -134,61 +184,97 @@ function CreateProductCategory(props: CreateProductCategoryProps) {
           <div className="mb-8 flex justify-between gap-6">
             <InputField
               required
-              label={t("modals-name", "Name")}
+              label={t("modals-name", "Name") as string}
               type="string"
-              name="name"
-              value={name}
               className="w-[338px]"
-              placeholder={t(
-                "modals-give-this-category-a-name",
-                "Give this category a name"
-              )}
-              onChange={(ev) => setName(ev.target.value)}
+              placeholder={
+                t(
+                  "modals-give-this-category-a-name",
+                  "Give this category a name"
+                ) as string
+              }
+              {...register("name", { required: true })}
             />
 
             <InputField
-              label={t("modals-handle", "Handle")}
+              label={t("modals-handle", "Handle") as string}
               type="string"
-              name="handle"
-              value={handle}
               className="w-[338px]"
-              placeholder={t("modals-custom-handle", "Custom handle")}
-              onChange={(ev) => setHandle(ev.target.value)}
+              placeholder={
+                t("modals-custom-handle", "Custom handle") as string
+              }
+              {...register("handle")}
             />
           </div>
 
           <div className="mb-8">
             <TextArea
               label={t("modals-description", "Description")}
-              name="description"
-              value={description}
-              placeholder={t(
-                "modals-give-this-category-a-description",
-                "Give this category a description"
-              )}
-              onChange={(ev) => setDescription(ev.target.value)}
+              placeholder={
+                t(
+                  "modals-give-this-category-a-description",
+                  "Give this category a description"
+                ) as string
+              }
+              {...register("description")}
             />
           </div>
 
           <div className="mb-8 flex justify-between gap-6">
             <div className="flex-1">
-              <NextSelect
-                label={t("modals-status", "Status")}
-                options={statusOptions(t)}
-                value={statusOptions(t)[isActive ? 0 : 1]}
-                onChange={(o) => setIsActive(o.value === "active")}
+              <Controller
+                name={"is_active"}
+                control={control}
+                rules={{ required: true }}
+                render={({ field }) => {
+                  return (
+                    <NextSelect
+                      {...field}
+                      label={t("modals-status", "Status") as string}
+                      placeholder="Choose a country"
```

Comment:
Change this to something that makes sense, such as `Choose visibility`, and the value should also be wrapped in t().